controllers/kustomization_controller.go

```diff
@@ -147,12 +147,7 @@ func (r *KustomizationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 
 	// resolve source reference
 	source, err := r.getSource(ctx, kustomization)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	// check source readiness
-	if source.GetArtifact() == nil {
+	if err != nil || source.GetArtifact() == nil {
```

Comment:
Looking at the logic of `getSource`, the result of this is that a "not found" or transient error (they are both the same there, instead of relying on `apierr.IsNotFound()`) is equal to the source not being ready. I am not sure if this is what you are after.

Reply:
Good point 👍 Please check the latest push.